docs/features/wait/exec.md

```diff
@@ -0,0 +1,19 @@
+# Exec Wait strategy
+
+The exec wait strategy will check the exit code of a process to be executed in the container, being able to set the following conditions:
```

Comment:
I am a little bit picky here, but I wonder how to consistently talk about _wait strategies_ in the docs. In Java, we are in a comfortable position, since the class name itself is fully descriptive within the domain language (e.g., `HttpWaitStrategy`, `LogMessageWaitStrategy`, etc.). So it follow the pattern of `XWaitStrategy`. 

In Go we generally have `XStrategy` (`ExecStrategy`, `HealthStrategy`) as types, but some are also off from the convention (`waitForSql`, which is also private?).

I know it would be a hard breaking change to align Go with Java here, but maybe we can eventually do it?

Reply:
In Go the package provides that context: `wait.ForLog`, `wait.ForSQL` etc, so we are covered there.

I totally agree that the existing code conventions are not consistent (public Vs private, reused fields such as startupTimeout and pollInterval being spread instead in of a separated struct or interface) and I did not change that intentionally. My goal is to face that in separate efforts